classification_analysis.ipynb

```diff
@@ -211,6 +230,49 @@
   {
    "cell_type": "code",
    "execution_count": null,
+   "id": "eb5df932",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pathlib import Path\n",
+    "\n",
+    "def get_reference_trajectory(trip_id: str, section_id: str):\n",
+    "\n",
+    "    root = Path(\"./bin/data\")\n",
+    "    return_file = None\n",
+    "\n",
+    "    workdir = root\n",
+    "    if (workdir / trip_id).exists():\n",
+    "        workdir = workdir / trip_id\n",
+    "        assert workdir.is_dir(), f\"{trip_id} is a file, not a dir.\"\n",
+    "        if (workdir / section_id).exists():\n",
+    "            workdir = workdir / section_id\n",
+    "            assert workdir.is_dir(), f\"{trip_id}.{section_id} is a file, not a dir.\"\n",
+    "            files = [f for f in workdir.iterdir()]\n",
+    "            assert files is not None and len(files) > 0, f\"No files found for {trip_id=} and {section_id=}\"\n",
+    "\n",
+    "            # sort the files by run number (just in case).\n",
+    "            files = list(sorted(files, key=lambda x: int(x.name.split(\"_\")[-1])))\n",
+    "\n",
+    "            # Concatenate all the trips into a single consolidated df but add an extra column that allows you\n",
+    "            # to filter on the run number (if required).\n",
+    "            df = pd.read_csv(files[0])\n",
+    "            df['run'] = '0'\n",
+    "\n",
```

Comment:
Ah I see that you are concatenating the entries from all the runs. That should work, but you should also be able to pass in the run_id for simplicity. then you don't need to iterate over all the files and can just read one file.
Is there a reason I am missing for why this does not work?

Reply:
Ah I see it is because the other ground truth is not run-specific. I am fine with this at the high level and you can resolve this comment after reading it.